### PR TITLE
Staging for version increment automation

### PR DESCRIPTION
**Pull request by prudhvigodithi**
Signed-off-by: pgodithi <pgodithi@amazon.com>

### Description
This is the staging PR for to execute workflows that auto raise the version increment PR's
Example: https://github.com/prudhvigodithi/ml-commons/pull/2
 
### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/1375
From solution: https://github.com/opensearch-project/opensearch-build/issues/1375#issuecomment-1159476686
Related issue: https://github.com/opensearch-project/ml-commons/issues/357
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Renaming to task updateVersion following https://github.com/opensearch-project/opensearch-plugin-template-java/pull/32.

File: build.gradle [hunk 1]
```diff
@@ -69,3 +69,20 @@ ext {
 dependencies {
     implementation 'junit:junit:4.12'
 }
+
+// updateVersion: Task to auto increment to the next development iteration
+task updateVersion {
+    onlyIf { System.getProperty('newVersion') }
```
Comment (ylwu-amzn) [inline build.gradle:75]: Where this `newVersion` will be set ? Where and when will this update version task will be executed?

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Just ask some question to learn more detail.

File: build.gradle (hunk 1 above)
Comment (prudhvigodithi) [reply build.gradle:75]: Example PR: https://github.com/prudhvigodithi/ml-commons/pull/2/files

**Comment by prudhvigodithi** [comment]
Hey @ylwu-amzn just checking back, if we are ok with PR, can we merge this?
Thank you

**Review by jackiehanyang — APPROVED** [review]

**Comment by ylwu-amzn** [comment]
> Hey @ylwu-amzn just checking back, if we are ok with PR, can we merge this? Thank you

Sure, merged. Thanks !

**Pull request merged**